### ISSUE TITLE
Flux Klein 9b training collapse

### ISSUE THREAD

**Issue by paul-lupu** [ostris/ai-toolkit#654]
hello, does training the 9b klein work? I ran this toml and the model completely collapsed by step 250: 
https://pastebin.com/yH60pPWp

by step 500 it blew up completely

**Comment by iamrohitanshu** [comment]
Aren't you training on the distilled model though?
Try the [base model](https://huggingface.co/black-forest-labs/FLUX.2-klein-base-9B/tree/main).

**Comment by Alrighty321** [comment]
> Aren't you training on the distilled model though? Try the [base model](https://huggingface.co/black-forest-labs/FLUX.2-klein-base-9B/tree/main).

Another question. I trained on the 9b base but its the only one listed on ostris aitoolkit. How can a I train a lora on the distilled? The lora I made isnt that good when using the distilled workflows

**Comment by iamrohitanshu** [comment]
@Alrighty321 I'm not an expert, but you might need to train your lora a bit more. Or you can increase the learning rate and see if it helps.
If you want to train on the distilled model, just download the distilled safetensors and put the path to it in the UI.

**Comment by orzechowy3334-rgb** [comment]
I give up! The model losses correct anatomy very fast. Waste of time. Flux.2 Klein won't be a star :(

**Comment by Illynir** [comment]
Yean same problem here, tested many settings and different configuration. The training goes badly from the 250 steps onwards, it starts to collapse.
And the more steps you add, the worse it gets.

For comparison, I have no training problems on other models, including Z image turbo, with the same datasets.

Not sure what's going on, whether it's the training tools that aren't quite ready for Klein yet, or whether it's Klein itself whch isn't suited for training.

Tested on Klein 9B Base of course.

**Comment by orzechowy3334-rgb** [comment]
I have recent experience with QWEN IMAGE EDIT2511 and Flux.2 Klein 9B: same datasets, same parameters. While Qwen reveals some adjustments to the concept with each epoch and a slow but steady decrease in losses, Flux.2 Klein reveals complete chaos in the control images and no loss gradient whatsoever. The latest parameters were as follows: RANK32, Sigmoid, LR = 0.0002, batch size = 1, experimental hints = 4.

**Comment by slightlyoutofphase** [comment]
IDK, CivitAI recently started using AI Toolkit for Klein training and after testing it I suspect the issue is with AI Toolkit, not the Klein models, as the results I got were 1000x worse than what I got testing the exact same dataset on ModelScope and Fal.ai, neither of which use AI Toolkit for Klein training in their backends.

**Comment by slightlyoutofphase** [comment]
> IDK, CivitAI recently started using AI Toolkit for Klein training and after testing it I suspect the issue is with AI Toolkit, not the Klein models, as the results I got were 1000x worse than what I got testing the exact same dataset on ModelScope and Fal.ai, neither of which use AI Toolkit for Klein training in their backends.

You can add TensorArt to the list of sites that don't use Toolkit and gave me great results, as of today

**Comment by jaretburkett** [comment]
This should be resolved in the latest version.
